New oom sync

mlaventure opened the pull request:
- Update containerd to use the new runc command `create/start`
- Remove direct dependency on runc state folder.

Depends on #256

mlaventure commented:
So I have no idea why the test fails on CI, I can't reproduce this timeout locally :/

crosbymichael commented:
When testing start of concurrent containers I get some weird logs and OOM errors that the memory cgroup cannot be found. 

https://gist.github.com/369f8426d3829a0534d03a1e2a610fdc

This usually happens when the code is a little racy and high stress causing the cgroups not to be there in time.

crosbymichael commented:
I'm using the benchmark tool in the `hack/benchmark.go`.  Just build it and
`sudo ./benchmark -count 20` you will see the issues

mlaventure commented:
Still trying to figure out why it's refusing to update the memory limit actually, which explains the extra logs :).

mlaventure commented:
I'll check the benchmark afterwards. But it should not be racy, since the user process shouldn't start before the signal is sent, and runc create should only return once everything is ready.

mlaventure commented:
Fixed the test. The Jenkins environment is requiring a lot more memory for the initial container env than my machine. Changed the test to have a memory hungry shell.

I can't reproduce your issue with the benchmark.

mlaventure commented:
Updated to get container's cgroup root from `/proc/<pid>/cgroup`

crosbymichael commented:
LGTM
